xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitleParser.h

```diff
@@ -26,6 +26,7 @@ class CDVDSubtitleParser
   virtual void Dispose() = 0;
   virtual void Reset() = 0;
   virtual CDVDOverlay* Parse(double iPts) = 0;
+  virtual std::string GetName() = 0;
```

Comment:
```suggestion
  virtual std::string GetName() const = 0;
```

Reply:
suggested change not compile, always say that cannot create all parsers classes without specify the reason
instead use `virtual const std::string GetName() = 0;`
works

Reply:
please return `const std::string&` as reference, also for the overridden function. never return plain const (one rule that sunk into me)

Reply:
Tried i have changed line 29 and 69 as:
`virtual const std::string& GetName() = 0;`
`std::string& const GetName() { return parserName; };`

but compiler complain
```
D:\GIT\kodi\xbmc\cores\VideoPlayer\DVDSubtitles\DVDSubtitleParser.h(69,29): warning C4227: anachronism used: qualifiers
 on reference are ignored (compiling source file D:\GIT\kodi\xbmc\cores\VideoPlayer\DVDSubtitles\DVDSubtitleParserSSA.c
pp) [D:\kodi-build\libkodi.vcxproj]
```

Reply:
i think on your 2nd line the & is misplaced, it should look like `std::string const& GetName() override { return parserName; };`

or even better
```c
const std::string& GetName() override { return parserName; };
```

Reply:
thanks now it's ok